x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformNodes.java

```diff
@@ -245,7 +249,24 @@ public static boolean nodeCanRunThisTransform(
             return false;
         }
 
+        if (isNodeShuttingDown(metadata, node.getId())) {
+            if (explain != null) {
+                explain.put(node.getId(), "node is shutting down");
+            }
+            return false;
+        }
+
         // we found no reason that the transform can not run on this node
         return true;
     }
+
+    private static boolean isNodeShuttingDown(Metadata metadata, String nodeId) {
```

Comment:
Perhaps this should be a utility method in node shutdown. I assume we need this (or may already have it) in other code that uses tasks and needs better explanations about why tasks are rejected.